### PR TITLE
feat: NEXUS-40875 - use nxrm-ha helm chart in operator-nexus-repository

### PR DESCRIPTION
**Pull request by jbermudezsonatype**
### Changes description
(🚀 Feature)
use nxrm-ha helm chart in operator-nexus-repository

### Context
The Nexus Repository open shift operator: operator-nexus-repository should use the same helm chart as nxrm-ha helm chart.

### Ticket ID
- https://sonatype.atlassian.net/browse/NEXUS-40875

### CI / Testing
TBD

### CODE REVIEW

File: helm-charts/nxrm-ha/templates/namespaces.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{{- if .Values.namespaces.nexusNs.enabled }}
```
Comment (jbermudezsonatype) [inline helm-charts/nxrm-ha/templates/namespaces.yaml:1]: Should this file be kept?

File: helm-charts/nxrm-ha/tests/namespaces_test.yaml [hunk 2]
```diff
@@ -0,0 +1,76 @@
+suite: test namespaces
```
Comment (jbermudezsonatype) [inline helm-charts/nxrm-ha/tests/namespaces_test.yaml:1]: Should this file be kept?

File: helm-charts/nxrm-ha/values.yaml [hunk 3]
```diff
@@ -1,84 +1,284 @@
...
----
+# Declare variables to be passed into your templates.
+namespaces:
+  nexusNs:
+    enabled: true
+    name: "nexusrepo"
+  cloudwatchNs:
+    enabled: false
+    name: amazon-cloudwatch
+  externaldnsNs: nexus-externaldns
```
Comment (jbermudezsonatype) [inline helm-charts/nxrm-ha/values.yaml:9]: Should these values be kept?

File: scripts/templates/sonatype.com_v1alpha1_nexusrepo_cr.yaml [hunk 4]
```diff
@@ -6,83 +6,229 @@ metadata:
...
+    hostPath: /
+    ingressClassName: null
     name: nexus-ingress
-    tls:
+  logStorage:
+    combineTaskLogs: true
+    tailSecondaryLogs: true
+  namespaces:
+    cloudwatchNs:
       enabled: false
```
Comment (jbermudezsonatype) [inline scripts/templates/sonatype.com_v1alpha1_nexusrepo_cr.yaml:45]: Should these values be kept?

File: deploy/crds/sonatype.com_v1alpha1_nexusrepo_cr.yaml [hunk 5]
```diff
@@ -6,83 +6,231 @@ metadata:
...
+    hostPath: /
+    ingressClassName: null
     name: nexus-ingress
-    tls:
+  logStorage:
+    combineTaskLogs: true
+    tailSecondaryLogs: true
+  namespaces:
+    cloudwatchNs:
       enabled: false
```
Comment (jbermudezsonatype) [inline deploy/crds/sonatype.com_v1alpha1_nexusrepo_cr.yaml:45]: Should these values be kept?

File: helm-charts/nxrm-ha/values.yaml [hunk 6]
```diff
@@ -1,84 +1,284 @@
-# DO NOT MODIFY. This is produced by template.
----
```
Comment (bobotimi) [inline helm-charts/nxrm-ha/values.yaml:2]: Please restore this line.

**Review by bobotimi — COMMENTED** [review]
Nice 1

File: helm-charts/nxrm-ha/values.yaml (hunk 3 above)
Comment (bobotimi) [reply helm-charts/nxrm-ha/values.yaml:10]: No this can go since we are not using it

**Review by bobotimi — COMMENTED** [review]
@jbermudezsonatype As the updates to the metadata are generated using the openshift tools, I'd says the next step in this is to test it in an openshift environment. The last time we updated the helm charts used by the operator @vswaminathan777 set up a Redhat Open Shift cluster on AWS which was accessible locally. This allowed us to deploy to it using the scripts contained in this repo. That's what we need at this point

**Review by bobotimi — APPROVED** [review]
Just need to rebase and test this in an openshift environment

**Pull request closed without merging**